test/request/hostname.js

```diff
@@ -11,6 +11,12 @@ describe('req.hostname', function(){
     req.hostname.should.equal('foo.com');
   })
 
+  it('should work with IPv6 Host', function(){
+    var req = request();
+    req.header.host = '[::1]:3000';
+    assert.equal(req.hostname, '[::1]');
+  })
+
```

Comment:
`::1` is the short hand for the main loopback address (the IPv6 version of 127.0.0.1).